controllers/pkg/reconcilers/bootstrap-packages/reconciler.go

```diff
@@ -88,36 +95,51 @@ type reconciler struct {
 
 func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	r.l = log.FromContext(ctx)
-
-	cr := &corev1.Secret{}
+	cr := &porchv1alpha1.PackageRevision{}
 	if err := r.Get(ctx, req.NamespacedName, cr); err != nil {
-		// if the resource no longer exists the reconcile loop is done
+		// There's no need to requeue if we no longer exist. Otherwise we'll be
+		// requeued implicitly because we return an error.
 		if resource.IgnoreNotFound(err) != nil {
 			msg := "cannot get resource"
 			r.l.Error(err, msg)
 			return ctrl.Result{}, errors.Wrap(resource.IgnoreNotFound(err), msg)
 		}
-		return reconcile.Result{}, nil
+		return ctrl.Result{}, nil
 	}
 
-	// if the secret is being deleted dont do anything for now
-	if cr.DeletionTimestamp != nil {
-		return reconcile.Result{}, nil
+	// check if the packagerevision is part of a staging repository
+	// if not we can ignore this package revision
+	stagingPR, err := r.IsStagingPackageRevision(ctx, cr)
```

Comment:
I think we should check the package lifecycle to make sure it  is Published (there is an IsPublished method in PR I believe, as "Deletion Proposed" is also considered published.

Reply:
fixed